### ISSUE TITLE
VIIRS AWIPS writer issues

### ISSUE THREAD

**Issue by kathys** [ssec/polar2grid#491]
Working with VIIRS data and the AWIPS writer, I found a couple of issues.

- True and false color imagery is not accepted by AWIPS.  But you were aware of this already.
- There is a navigation pixel shift of at least one pixel to the southeast in the new writer.  It might actually be more accurate.  Scott L. and I will have to look more closely.
- The spatial resolution of the files looks the same in both.  Excellent.
- Band 2 :physical_element = "0.86 um" ; needs to be changed to :physical_element = "0.87 um" otherwise the AWIPS menus will not recognize it.
- There is gray at the tile boundaries produced by the P2G V3.0 AWIPS writer.  Is there flag that I should be using to prevent this?  See example below.

![Image](https://user-images.githubusercontent.com/1848019/181104490-cb88d569-61c3-434d-aa8a-9a1eaa2f8688.png)

**Comment by djhoese** [comment]
Regarding the gray regions, Are these the tiles that didn't have the proper attributes so AWIPS wasn't recognizing them and didn't know how to scale them properly?

**Comment by kathys** [comment]
No, it was happening for files that VIIRS recognized without having to do any changes to them.  Such as VIIRS I-Band 1, 3, 4, 5 and M-band 15 (shown in the image above).  The files created for these individual bands using P2G V3.0 displayed identically in every other way besides these gray borders, and the pixel shift.

**Comment by djhoese** [comment]
Just to make sure, could you paste the command you used to generate these files?

**Comment by kathys** [comment]
All of the commands look like this:

polar2grid.sh -r viirs_sdr -w awips_tiled --num-workers 4 -p i01 -g lcc_conus_300 --sector-id LCC --letters --compress -f input

**Comment by kathys** [comment]
Both input SDR data sources and AWIPS output tiles are available here:

`bumi:/data/users/kathys/viirs_awips_2022`

**Comment by djhoese** [comment]
Here's the i01 data variable for v2.3:

```
        short data(y, x) ;
                data :_FillValue = 32767s ;
                data :coordinates = "y x" ;
                data :scale_factor = 4.436114e-05f ;
                data :add_offset = 0.f ;
                data :units = "*1" ;
                data :valid_min = 0s ;
                data :valid_max = 32766s ;
                data :standard_name = "toa_bidirectional_reflectance" ;
                data :grid_mapping = "lambert_projection" ;
```

And for 3.0:

```
        ushort data(y, x) ;
                data :_FillValue = 65535US ;
                data :physical_element = "0.64 um" ;
                data :units = "%" ;
                data :grid_mapping = "lambert_projection" ;
                data :coordinates = "y x" ;
                data :add_offset = 0.f ;
                data :scale_factor = 0.002228013f ;
```

So it was what I feared: AWIPS is probably being picky about unsigned versus signed integers. I will try adding the `valid_range` and see if AWIPS likes that better. I kind of want to read the source code because this is...dumb.

**Comment by djhoese** [comment]
This was fixed as part of #492 and #493.
